apps/train_loc/lib/train_loc/encoder/vehicle_positions_enhanced.ex

```diff
@@ -46,6 +79,8 @@ defmodule TrainLoc.Encoder.VehiclePositionsEnhanced do
 
   defp build_entity(_), do: []
 
+  @spec entity_trip(Vehicle.t()) ::
+          %{start_date: String.t()} | %{start_date: String.t(), trip_short_name: String.t()}
```

Comment:
suggestion: since you're re-using this from the `entity` spec, maybe pull it out and give it a name to use here?

Reply:
Yeah I had these as `entity_trip` and `entity_vehicle` at one point and reverted them. Good idea.